src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -3189,7 +3189,10 @@ class ExecuteScriptSegment(BaseSegment):
     match_grammar = Sequence(
         OneOf("EXEC", "EXECUTE"),
         Sequence(Ref("ParameterNameSegment"), Ref("EqualsSegment"), optional=True),
-        Ref("ObjectReferenceSegment"),
+        OneOf(
+            Ref("ObjectReferenceSegment"),
+            Bracketed(Ref("ObjectReferenceSegment")),
+        ),
```

Comment:
Did you try this?

```suggestion
        OptionallyBracketed(
            Ref("ObjectReferenceSegment"),
        ),
```

Sometimes causes problems so sometimes need what you had, but think you should be able to use it here.

Reply:
Thank you. Searched for it and eventually found this less concise expression here:
https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_tsql.py#L1263

Reply:
Yeah that's one that may need the other syntax as you can't provide a `bracketed_type` to `OptionallyBracketed` but that's not the case for your one.